cmd/calyptia/create_core_instance_aws.go

```diff
@@ -1,19 +1,203 @@
 package main
 
 import (
-	"errors"
+	"context"
+	"encoding/json"
+	"fmt"
+	"time"
 
+	"github.com/calyptia/api/types"
+	awsClient "github.com/calyptia/cli/aws"
+	"github.com/pkg/errors"
+	"github.com/sethvargo/go-retry"
 	"github.com/spf13/cobra"
 )
 
-func newCmdCreateCoreInstanceOnAWS(config *config) *cobra.Command {
+const (
+	coreInstanceUpCheckTimeout = 10 * time.Minute
+	coreInstanceUpCheckBackoff = 5 * time.Second
+)
+
+var (
+	errCoreInstanceNotFound        = errors.New("core instance not found")
+	errCoreInstanceNotRunning      = errors.New("core instance not in running status")
+	coreInstanceUpCheckMaxDuration = func() retry.Backoff {
+		return retry.WithMaxDuration(coreInstanceUpCheckTimeout, retry.NewConstant(coreInstanceUpCheckBackoff))
+	}
+)
+
+func newCmdCreateCoreInstanceOnAWS(config *config, client awsClient.Client) *cobra.Command {
+	var (
+		tags                  []string
+		noHealthCheckPipeline bool
+		coreInstanceVersion   string
+		coreInstanceName      string
+		environmentKey        string
+		credentials           string
+		profileFile           string
+		profileName           string
+		region                string
+		subnetID              string
+		keyName               string
+		instanceTypeName      string
+		securityGroupName     string
+	)
+
 	cmd := &cobra.Command{
 		Use:     "aws",
 		Aliases: []string{"ec2", "amazon"},
-		Short:   "Setup a new core instance on Amazon EC2 (TODO)",
+		Short:   "Setup a new core instance on Amazon EC2",
 		RunE: func(cmd *cobra.Command, args []string) error {
-			return errors.New("not implemented")
+			var err error
+			var params awsClient.CreateInstanceParams
+
+			ctx := context.Background()
+
+			if client == nil {
+				client, err = awsClient.New(ctx, region, credentials, profileFile, profileName)
+				if err != nil {
+					return errors.Wrap(err, "could not initialize client")
```

Comment:
Let's just stick to standard error package:
```suggestion
					return fmt.Errorf("could not initialize client: %w", err)
```